src/app/components/secret-key/mnemonic-key/mnemonic-input-field.tsx

```diff
@@ -0,0 +1,153 @@
+import { useState } from 'react';
+
+import { TextField } from '@radix-ui/themes';
+import { useField } from 'formik';
+import { css } from 'leather-styles/css';
+import { FlexProps, styled } from 'leather-styles/jsx';
+import { token } from 'leather-styles/tokens';
+
+import { useShowFieldError } from '@app/common/form-utils';
+
+interface InputFieldProps extends FlexProps {
+  dataTestId?: string;
+  name: string;
+  value: string;
+  onPaste: (e: React.ClipboardEvent<HTMLInputElement>) => void;
+  onUpdateWord: (word: string) => void;
+  hasError?: boolean;
+  wordlist: string[];
+}
+export function InputField({
+  dataTestId,
+  name,
+  onPaste,
+  onChange,
+  // onUpdateWord,
+  value,
+}: // wordlist,
+InputFieldProps) {
+  // const triggerButtonRef = useRef<HTMLButtonElement>(null);
+  const [field] = useField(name);
+  const [ttype, setType] = useState<'text' | 'password'>('password');
+  const [isFocused, setIsFocused] = useState(false);
+
+  // const buttonRef = useRef<HTMLButtonElement>(null);
+
+  // const [autoSuggestOpen, setAutoSuggestOpen] = useState(false);
+
+  // const [open, setOpen] = useState<boolean>();
+  // const regex = new RegExp('^' + field.value, 'i');
+
+  const showError = useShowFieldError(name);
+
+  // useEffect(() => {
+  //   // console.log(field.value);
+  //   if (field.value.length > 3) {
+  //     // triggerButtonRef.current.click();
+  //     buttonRef.current?.click();
+  //     document.getElementById('button')?.click();
+  //     // setOpen(true);
+  //   }
+  //   // setOpen(false);
+  // }, [field.value]);
+
+  return (
+    <div>
+      <TextField.Root
+        /**
+         * Focus styling:
+         * radix inserts a <div inside the root with .rt-TextFieldChrome
+         * onFocus - this adds an unwanted box-shadow
+         * setting color="brown" makes this transparent
+         * then focus border can be controlled more easily
+         */
+        color="brown"
+        className={css({
+          display: 'flex',
+          alignSelf: 'stretch',
+          height: '48px',
+          width: '176px',
+          gap: 'space.01',
+          px: 'space.03',
+          border: isFocused
+            ? `2px solid ${token('colors.accent.text-primary')}`
+            : showError
+            ? `2px solid ${token('colors.error')}`
+            : 'none',
```

Comment:
There is no `_focused` property on the radix `TextField.Input`, one of the drawbacks. Or else I didn't figure it out. 

I can dig deeper into it

Reply:
Thanks for the input here team. Kyran helped me figure this out by suggesting the `_focusWithin` property. 

I then used the `data-state` attribute to add a flag for when the form was in an error state or not. 

The `TextField` does accept `_focus` but my difficulty was controlling the border of `TextField.Root` via focus on the child `TextField.Input`. 

Now we have this figured out I think this could be used as a multi-use input field